QPong-Unity-Version/Assets/Scripts/CircuitGridClient.cs

```diff
@@ -27,70 +31,97 @@ void Start()
         GameObject CircuitGrid = GameObject.Find("CircuitGrid");
         CircuitGridControl CircuitGridControlScript = CircuitGrid.GetComponent<CircuitGridControl>();
         // paddleArray = CircuitGridControlScript.paddleArray;
-        var gateArrayString = string.Join(",", GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().gateArray);
-        StartCoroutine(getStatevector(gateArrayString));
+        GetStateVector(gateArrayString);
     }
 
+    //TODO: find out if there is ever an instance of both flags being true for one update, and do we need that to happen? can we optimize to just one?
```

Comment:
There won't be an instance of both flags being true. But I use getStatevectorFlag to signal sending request to python server if and only if there is change in the circuit. We can opt to send request in a regular interval, but because the communication between unity and python server is slow, the game will become laggy.

Reply:
ok i was thinking about how this works, and if one if true every time Update is called this would cause a lag.  How often does one of those flags get set to true and trigger?